packages/malabi/src/instrumentation/index.ts

```diff
@@ -46,3 +86,26 @@ export const instrument = () => {
         tracerProvider,
     });
 };
+
+/**
+ * A wrapper that handles creating a span per test run. returns the spans that were created ready for assertion.
+ * @category Main Functions
+ * @param callback an async function containing all of the current test's span generating operations(API calls etc)
+ */
+export const malabi = async (callback): Promise<TelemetryRepository> => {
+    return new Promise((resolve) => {
+        const tracer = trace.getTracer('malabiManualTracer');
+
+        tracer.startActiveSpan('malabiRoot', async (span) => {
+            const currTraceID = span.spanContext().traceId;
+            await callback();
```

Comment:
wrap in try catch and add a test to verify that exceptions inside the callback are propagated correcly outside the wrapper